### ISSUE TITLE
Release Windows binaries in zip archive instead of .tar.gz

### ISSUE THREAD

**Issue by mloskot** [kubernetes-sigs/kustomize#5116]
### Eschewed features

- [X] This issue is not requesting templating, unstuctured edits, build-time side-effects from args or env vars, or any other eschewed feature.

### What would you like to have added?

I have proposed package with `kustomize` to [winget - official package manager for Windows](https://learn.microsoft.com/en-gb/windows/package-manager/winget/), see https://github.com/microsoft/winget-pkgs/pull/101695

Unfortunately, winget only supports zip archive or raw exe as portable 'installer', see https://github.com/microsoft/winget-cli/issues/2899

So, the packaging of `kustomize` is currently blocked.

### Why is this needed?

There already are several Kubernetes toools packaged for winget: `kubectl` and `kompose`, and `minikube` too.
It would be awesome if the `kustomize` joined in and be offered to install on Windows with the simple `winget` command:

```console
winget install --id Kubernetes.kubectl
```

The other tools are available to download as plain `.exe` files:
- https://github.com/microsoft/winget-pkgs/blob/master/manifests/k/Kubernetes/kubectl/1.26.3/Kubernetes.kubectl.installer.yaml#L13
- https://github.com/microsoft/winget-pkgs/blob/a4cf5332206b4fc0237fbb482fed8e1a87a8b043/manifests/k/Kubernetes/kompose/1.28.0/Kubernetes.kompose.installer.yaml#L11

It would be very helpful if `kustomize.exe` was also available like those two.

### Can you accomplish the motivating task without this feature, and if so, how?

No.

### What other solutions have you considered?

None exist.

### Feature ownership

- [ ] I am interested in contributing this feature myself! 🎉

**Comment by mloskot** [comment]
I did a test using `kubectl.exe` 49 MB compressing it to 14 MB `.zip` and 14 MB `.gz` and 10 MB `.xy`
Yes, `.xy` offers best compression, but `.zip`  is already the download size improvement and it would allow packaging for winget.

**Comment by KnVerey** [comment]
/triage accepted
/priority important-longterm

It looks like there is a `format_overrides` option for the go-releaser configuration that would allow us to do this:

```yaml
    # Can be used to change the archive formats for specific GOOSs.
    # Most common use case is to archive as zip on Windows.
    format_overrides:
      - goos: windows
        format: zip
```

This would go in our go-releaser configuration here: https://github.com/kubernetes-sigs/kustomize/blob/master/releasing/run-goreleaser.sh#L109

And instructions for testing that configuration are here: https://github.com/kubernetes-sigs/kustomize/tree/master/releasing#testing-changes-to-the-release-pipeline

/assign @nathanmartins

**Comment by natasha41575** [comment]
This will need to wait for https://github.com/kubernetes-sigs/kustomize/issues/5220 to be completed first.

**Comment by natasha41575** [comment]
I just approved https://github.com/kubernetes-sigs/kustomize/pull/5239, which fixes #5520. It should merge soon. After it merges, if someone wants to add support for windows zip in the new release script, please feel free to take a stab at it.

**Comment by koba1t** [comment]
Hi @natasha41575 

> After it merges, if someone wants to add support for windows zip in the new release script, please feel free to take a stab at it.

I've [already added zip support on the windows release](https://github.com/kubernetes-sigs/kustomize/pull/5239/files#diff-f360f726ff58796695d8f13298f80887a97e6910b9d31da50f462192f16edc82R62) in the previous PR.
So, I believe resolved this issue by merging that PR.

Example: https://github.com/koba1t/kustomize/releases/tag/kustomize%2Fv5.1.2

**Comment by mikaello** [comment]
Why is this issue still open? It seems like Windows binaries have been released in zip archives since [Kustomize 5.2.0](https://github.com/kubernetes-sigs/kustomize/releases/tag/kustomize%2Fv5.2.0).

**Comment by mloskot** [comment]
Good point. I am closing this issue myself.

FYI, Kustomize 5.3.0 is now available via winget
- https://github.com/microsoft/winget-pkgs/pull/129954
